source/common/upstream/outlier_detection_impl.cc

```diff
@@ -456,100 +473,62 @@ void DetectorImpl::runCallbacks(HostSharedPtr host) {
   }
 }
 
-void EventLoggerImpl::logEject(HostDescriptionConstSharedPtr host, Detector& detector,
-                               EjectionType type, bool enforced) {
-  // TODO(mattklein123): Log friendly host name (e.g., instance ID or DNS name).
-  // clang-format off
-  static const std::string json_5xx =
-    std::string("{{") +
-    "\"time\": \"{}\", " +
-    "\"secs_since_last_action\": \"{}\", " +
-    "\"cluster\": \"{}\", " +
-    "\"upstream_url\": \"{}\", " +
-    "\"action\": \"eject\", " +
-    "\"type\": \"{}\", " +
-    "\"num_ejections\": {}, " +
-    "\"enforced\": \"{}\"" +
-    "}}\n";
-
-  static const std::string json_success_rate =
-    std::string("{{") +
-    "\"time\": \"{}\", " +
-    "\"secs_since_last_action\": \"{}\", " +
-    "\"cluster\": \"{}\", " +
-    "\"upstream_url\": \"{}\", " +
-    "\"action\": \"eject\", " +
-    "\"type\": \"{}\", " +
-    "\"num_ejections\": {}, " +
-    "\"enforced\": \"{}\", " +
-    "\"host_success_rate\": \"{}\", " +
-    "\"cluster_average_success_rate\": \"{}\", " +
-    "\"cluster_success_rate_ejection_threshold\": \"{}\"" +
-    "}}\n";
-  // clang-format on
-  SystemTime now = time_source_.systemTime();
-  MonotonicTime monotonic_now = time_source_.monotonicTime();
+void EventLoggerImpl::logEject(const HostDescriptionConstSharedPtr host, Detector& detector,
+                               envoy::data::cluster::v2alpha::OutlierEjectionType type,
+                               bool enforced) {
+  envoy::data::cluster::v2alpha::OutlierDetectionEvent event;
+  event.set_type(type);
 
-  switch (type) {
-  case EjectionType::Consecutive5xx:
-  case EjectionType::ConsecutiveGatewayFailure:
-    file_->write(fmt::format(
-        json_5xx, AccessLogDateTimeFormatter::fromTime(now),
-        secsSinceLastAction(host->outlierDetector().lastUnejectionTime(), monotonic_now),
-        host->cluster().name(), host->address()->asString(), typeToString(type),
-        host->outlierDetector().numEjections(), enforced));
-    break;
-  case EjectionType::SuccessRate:
-    file_->write(fmt::format(
-        json_success_rate, AccessLogDateTimeFormatter::fromTime(now),
-        secsSinceLastAction(host->outlierDetector().lastUnejectionTime(), monotonic_now),
-        host->cluster().name(), host->address()->asString(), typeToString(type),
-        host->outlierDetector().numEjections(), enforced, host->outlierDetector().successRate(),
-        detector.successRateAverage(), detector.successRateEjectionThreshold()));
-    break;
+  absl::optional<MonotonicTime> time = host->outlierDetector().lastUnejectionTime();
+  setCommonEventParams(event, host, time);
+
+  event.set_action(envoy::data::cluster::v2alpha::Action::EJECT);
+
+  event.set_enforced(enforced);
+
+  if (type == envoy::data::cluster::v2alpha::OutlierEjectionType::SUCCESS_RATE) {
+    event.mutable_eject_success_rate_event()->mutable_cluster_average_success_rate()->set_value(
```

Comment:
The way I'm reading this code, these values always get set for this message. Do the values have to be WKT in this case or can they just be normal types?